kiteconnect/ticker.py

```diff
@@ -613,9 +610,9 @@ def set_mode(self, mode, instrument_tokens):
                 six.b(json.dumps({"a": self._message_setmode, "v": [mode, instrument_tokens]}))
             )
 
-            # Update modes
-            for token in instrument_tokens:
-                self.subscribed_tokens[token] = mode
+            # update instrument token and mode
+            self.subscribe_mode = mode
+            self.subscribe_tokens = instrument_tokens
```

Comment:
You are just replacing the existing subscribed tokens with the incoming one which is wrong. Images this scenario 

```
ws.subscribe([1, 2, 3])
ws.setMode(ws.MODE_FULL, [1])
ws.subscribe(ws.MODE_LTP, [2,3])
```
Now the value of `self.subscribe_tokens` is `[2,3]`? which is wrong since there is one more subscription for token `1`. So when `resubscribe` is called it's going to just subscribe `[2,3]` and miss `1`.